src/main/java/io/codelex/flightplanner/flightmanager/FlightService.java

```diff
@@ -0,0 +1,114 @@
+package io.codelex.flightplanner.flightmanager;
+
+import io.codelex.flightplanner.flightmanager.domain.Airport;
+import io.codelex.flightplanner.flightmanager.domain.Flight;
+import io.codelex.flightplanner.flightmanager.domain.PageResult;
+import io.codelex.flightplanner.flightmanager.domain.SearchFlightRequest;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+@Service
+public class FlightService {
+    private final FlightRepo repository;
+
+    public FlightService(FlightRepo repository) {
+        this.repository = repository;
+    }
+
+    public void clearFlights() {
+        this.repository.clearFlights();
+    }
+
+    public void addFlight(Flight req) {
+        this.repository.addFlight(req);
+    }
+
+    public boolean isValidAirports(Flight req) {
+        return !req.getFrom().equals(req.getTo()) && this.repository.airportList.contains(req.getTo()) && this.repository.airportList.contains(req.getFrom());
+    }
+
+    public boolean isNotRepeated(Flight flight) {
+        boolean isNotRep = true;
+        int size = this.repository.flightList.stream().filter(f -> f.equals(flight)).toList().size();
```

Comment:
Var izmantot Stream iebūvēto anyMatch - uzreiz pārbauda vai kāds atbilst kritērijam un atgriež boolean, nav tad tas size jāpārbauda.